### PR TITLE
Resolving issue 34 on python-jobs-board.

### PR DESCRIPTION
**Pull request by SarahPythonista**
Adds listener to send email when new job is submitted for approval.

### CODE REVIEW

File: jobs/tests/test_models.py [hunk 1]
```diff
@@ -127,3 +128,13 @@ def test_get_previous_approved(self):
     def test_region_optional(self):
         job = self.create_job(region=None)
 
+    def test_email_on_submit(self):
+        job = self.create_job()
+        mail.outbox = []
```
Comment (berkerpeksag) [inline jobs/tests/test_models.py:143]: This line can be removed. Django's TestCase class already does this: https://docs.djangoproject.com/en/dev/topics/testing/tools/#emptying-the-test-outbox

File: jobs/tests/test_models.py [hunk 2]
```diff
@@ -127,3 +128,13 @@ def test_get_previous_approved(self):
...
+        self.assertEqual(0, len(mail.outbox))
```
Comment (berkerpeksag) [inline jobs/tests/test_models.py:144]: It would be good to test email subject too.

File: templates/jobs/email/job_was_submitted.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+The following job was submitted for approval:
+
+{{ content_object.display_name }}
+
+To view and review:
+
+URL:  https://{{ site.domain }}{{ content_object.get_absolute_url }}
```
Comment (berkerpeksag) [inline templates/jobs/email/job_was_submitted.txt]: Two whitespaces after "URL:".

**Comment by berkerpeksag** [comment]
Could you rebase your branch with the latest master? Thanks!

File: jobs/tests/test_models.py (hunk 1 above)
Comment (ubernostrum) [inline jobs/tests/test_models.py:143]: Actually that line is necessary. The listener is looking for a `Job` saved with its status equal to `STATUS_REVIEW`, and that's the _default_ value of the field. So the factory calls caused by `create_job()` will pollute the outbox, and it's necessary to clear it and then do a manual save and test.

Comment (berkerpeksag) [inline jobs/tests/test_models.py:143]: Thanks for the explanation. I think adding a comment would be helpful.

**Pull request merged**